relay-spans/src/v2_to_v1.rs

```diff
@@ -270,6 +277,178 @@ fn derive_op_for_v2_span(span: &SpanV2) -> String {
+/// thorough implementation for other types of spans for now.
+fn derive_description_for_v2_span(span: &SpanV2) -> Option<String> {
+    // `name` is a low-cardinality description of the span, so it makes for a good fallback.
+    let description = span.name.value().map(|v| v.to_owned());
+
+    let Some(attributes) = span.attributes.value() else {
+        return description;
+    };
+
+    if attributes.contains_key("http.request.method") || attributes.contains_key("http.method") {
+        if let Some(http_description) = derive_http_description(attributes, &span.kind.value()) {
+            return Some(http_description);
+        }
+    }
+
+    if attributes.contains_key("db.system") || attributes.contains_key("db.system.name") {
+        let is_cache_op = attributes
+            .get("sentry.op")
+            .and_then(|attr| attr.value())
+            .and_then(|attr_val| attr_val.value.value.value())
+            .and_then(|v| v.as_str())
+            .map(|op| op.starts_with("cache."))
+            .unwrap_or(false);
+
+        if !is_cache_op {
+            if let Some(database_description) = derive_db_description(attributes) {
+                return Some(database_description);
+            }
+        }
+    }
+
+    description
+}
+
+fn derive_http_description(
+    attributes: &Object<Attribute>,
+    kind: &Option<&SpanV2Kind>,
+) -> Option<String> {
+    // Get HTTP method
+    let http_method = attributes
+        .get("http.request.method")
+        .or_else(|| attributes.get("http.method"))
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())?;
+
+    let description = http_method.to_owned();
+
+    // Get URL path information
+    let url_path = match kind {
+        Some(SpanV2Kind::Server) => get_server_url_path(attributes),
+        Some(SpanV2Kind::Client) => get_client_url_path(attributes),
+        _ => None,
+    };
+
+    if url_path.is_none() {
+        return Some(description);
+    }
+
+    let url_path = url_path.unwrap();
+    let base_description = format!("{} {}", http_method, url_path);
+
+    // Check for GraphQL operations
+    if let Some(graphql_ops) = attributes
+        .get("sentry.graphql.operation")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(format!("{} ({})", base_description, graphql_ops));
+    }
+
+    Some(base_description)
+}
+
+fn derive_db_description(attributes: &Object<Attribute>) -> Option<String> {
+    // `db.query.text` is a recommended attribute, and it contains the full query text if available.
+    // This is the ideal description.
+    if let Some(query_text) = attributes
+        .get("db.query.text")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(query_text.to_string());
```

Comment:
We generally use `to_owned` for this.
```suggestion
        return Some(query_text.to_owned());
```